include/espi.h

```diff
@@ -0,0 +1,769 @@
+	ESPI_CYCLE_OKCOMPLETION_DATA,
+	ESPI_CYCLE_NOK_COMPLETION_NODATA,
+};
+
+/**
+ * @brief eSPI system platform signals that can be send or receive through
+ * virtual wire channel
+ */
+enum espi_vwire_signal {
+	/* Virtual wires that can only be send from master to slave */
+	ESPI_VWIRE_SIGNAL_SLP_S3,
+	ESPI_VWIRE_SIGNAL_SLP_S4,
+	ESPI_VWIRE_SIGNAL_SLP_S5,
+	ESPI_VWIRE_SIGNAL_OOB_RST_WARN,
+	ESPI_VWIRE_SIGNAL_PLTRST,
+	ESPI_VWIRE_SIGNAL_SUS_STAT,
+	ESPI_VWIRE_SIGNAL_NMIOUT,
+	ESPI_VWIRE_SIGNAL_SMIOUT,
+	ESPI_VWIRE_SIGNAL_HOST_RST_WARN,
+	ESPI_VWIRE_SIGNAL_SLP_A,
+	ESPI_VWIRE_SIGNAL_SUS_PWRDN_ACK,
+	ESPI_VWIRE_SIGNAL_SUS_WARN,
+	ESPI_VWIRE_SIGNAL_SLP_WLAN,
+	ESPI_VWIRE_SIGNAL_SLP_LAN,
+	ESPI_VWIRE_SIGNAL_HOST_C10,
+	ESPI_VWIRE_SIGNAL_DNX_WARN,
+	/* Virtual wires that can only be send from slave to master */
+	ESPI_VWIRE_SIGNAL_PME,
+	ESPI_VWIRE_SIGNAL_WAKE,
+	ESPI_VWIRE_SIGNAL_OOB_RST_ACK,
+	ESPI_VWIRE_SIGNAL_SLV_BOOT_STS,
+	ESPI_VWIRE_SIGNAL_ERR_NON_FATAL,
+	ESPI_VWIRE_SIGNAL_ERR_FATAL,
+	ESPI_VWIRE_SIGNAL_SLV_BOOT_DONE,
+	ESPI_VWIRE_SIGNAL_HOST_RST_ACK,
+	ESPI_VWIRE_SIGNAL_RST_CPU_INIT,
+	/* System management interrupt */
+	ESPI_VWIRE_SIGNAL_SMI,
+	/* System control interrupt */
+	ESPI_VWIRE_SIGNAL_SCI,
+	ESPI_VWIRE_SIGNAL_DNX_ACK,
+	ESPI_VWIRE_SIGNAL_SUS_ACK,
+};
+
+/**
+ * @brief eSPI event
+ */
+struct espi_event {
+	/** Event type */
+	enum espi_bus_event evt_type;
+	/** Additional details for bus event type */
+	u32_t evt_details;
+	/** Data associated to the event */
+	u32_t evt_data;
+};
+
+/**
+ * @brief eSPI bus configuration parameters
+ */
+struct espi_cfg {
+	/** Supported I/O mode */
+	enum espi_io_mode io_caps;
+	/** Supported channels */
+	enum espi_channel channel_caps;
+	/** Maximum supported frequency in MHz*/
+	u8_t max_freq;
+};
+
+/**
+ * @brief eSPI peripheral request packet format
+ */
+struct espi_request_packet {
+	enum espi_cycle_type cycle_type;
+	u8_t tag;
+	u16_t len;
+	u32_t address;
+	u8_t *data;
+};
+
+/**
+ * @brief eSPI out-of-band transaction packet format
+ */
+struct espi_oob_packet {
+	u8_t *buf;
+	u16_t len;
+};
+
+/**
+ * @brief eSPI flash transactions packet format
+ */
+struct espi_flash_packet {
+	u8_t *buf;
+	u32_t flash_addr;
+	u16_t len;
+};
+
+struct espi_callback;
+
+/**
+ * @typedef espi_callback_handler_t
+ * @brief Define the application callback handler function signature.
+ *
+ * @param dev Device struct for the eSPI device.
+ * @param cb Original struct espi_callback owning this handler.
+ * @param espi_evt event details that trigger the callback handler.
+ *
+ */
+typedef void (*espi_callback_handler_t)(struct device *dev,
+					struct espi_callback *cb,
+					struct espi_event espi_evt);
+
+/**
+ * @cond INTERNAL_HIDDEN
+ *
+ * Used to register a callback in the driver instance callback list.
+ * As many callbacks as needed can be added as long as each of them
+ * are unique pointers of struct espi_callback.
+ * Beware such structure should not be allocated on stack.
```

Comment:
I think this line is wrong.

Reply:
Sounds fine to me. Callback structures have to be statically allocated (as for gpio callbacks, etc...) It's worth noticing it in the documentation.